smartmeter/smartmeter.ino

```diff
@@ -47,12 +51,18 @@ float readCurrent()
 {
     static float current = 0;
 
-    Data::self().data->actualCurrent = (analogRead(A0)-450.5)/450.5*30.0;
+    Data::self().data->actualCurrent = (analogRead(CURRENT_SENSOR)-450.5)/450.5*30.0;
     current = current*0.95 + 0.05*Data::self().data->actualCurrent;
     Data::self().data->averageCurrent = current;
     return current;
 }
 
+void changeButtonState()
+{
+    buttonState = !buttonState;
+    Data::self().data->buttonState = buttonState;
```

Comment:
could you just do 
Data::self().data->buttonState = !Data::self().data->buttonState;
instead?